sklearn/decomposition/_nmf.py

```diff
@@ -1347,6 +1347,9 @@ def transform(self, X):
             Transformed data.
         """
         check_is_fitted(self)
+        X = self._validate_data(X, accept_sparse=('csr', 'csc'),
+                                dtype=[np.float64, np.float32],
+                                reset=False)
 
         W, _, n_iter_ = non_negative_factorization(
```

Comment:
`non_negative_factorization` also does a call to `check_array` internally so calling `_validate_data` here causes some performance overhead. For simplicity's sake I don't want to do optimize this as part of this PR but we might want to add a kwarg to `non_negative_factorization`  to skip input validation.

Reply:
maybe write that as a TODO comment?

Reply:
This is most likely going to appear in other places. Another solution would be to have a private `_non_negative_factorization` that does not call `check_array`, but still check for non-negative values.

Reply:
I added a comment.